zebrad/tests/acceptance.rs

```diff
@@ -3692,6 +3695,93 @@ async fn has_spending_transaction_ids() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test(flavor = "multi_thread")]
+async fn invalidate_and_reconsider_block() -> Result<()> {
+    use std::sync::Arc;
+
+    use common::regtest::MiningRpcMethods;
+    use zebra_chain::parameters::testnet::REGTEST_NU5_ACTIVATION_HEIGHT;
+
+    let _init_guard = zebra_test::init();
+    let mut config = os_assigned_rpc_port_config(false, &Network::new_regtest(Default::default()))?;
+    config.state.ephemeral = false;
+
+    let test_dir = testdir()?.with_config(&mut config)?;
+
+    let mut child = test_dir.spawn_child(args!["start"])?;
+    let rpc_address = read_listen_addr_from_logs(&mut child, OPENED_RPC_ENDPOINT_MSG)?;
+
+    tracing::info!("waiting for Zebra state cache to be opened");
+
+    tokio::time::sleep(LAUNCH_DELAY).await;
+
+    let rpc_client = RpcRequestClient::new(rpc_address);
+    let mut blocks = Vec::new();
+    for _ in 0..50 {
+        let (block, _) = rpc_client
+            .block_from_template(Height(REGTEST_NU5_ACTIVATION_HEIGHT))
+            .await?;
+
+        rpc_client.submit_block(block.clone()).await?;
+        blocks.push(block);
+    }
+
+    tracing::info!("checking that read state has the new non-finalized best chain blocks");
+    for expected_block in blocks.clone() {
+        let height = expected_block.coinbase_height().unwrap();
+        let zebra_block = rpc_client
+            .get_block(height.0 as i32)
+            .await
+            .map_err(|err| eyre!(err))?
+            .expect("Zebra test child should have the expected block");
+
+        assert_eq!(
+            zebra_block,
+            Arc::new(expected_block),
```

Comment:
If the compiler doesn't optimize the assert, then dereferencing the arc should be faster than allocating the expected block on the heap.

```suggestion
            *zebra_block, expected_block,
```

Reply:
Good point, noted. I'd be happy to apply this, but since it's a test and the code is only called a handful of times in the test, I think the difference is insignificant and I'd prefer to merge this as-is if we don't need to make changes for the other comment.